subsys/mgmt/hawkbit/hawkbit.c

```diff
@@ -778,8 +784,16 @@ int hawkbit_default_config_data_cb(const char *device_id, uint8_t *buffer, const
 #ifdef CONFIG_HAWKBIT_SET_SETTINGS_RUNTIME
 int hawkbit_set_config(struct hawkbit_runtime_config *config)
 {
+	size_t length;
+
 	if (k_sem_take(&probe_sem, HAWKBIT_SET_SERVER_TIMEOUT) == 0) {
 		if (config->server_addr != NULL) {
+			length = strnlen(config->server_addr, sizeof(hb_cfg.server_addr));
+			if (length > sizeof(hb_cfg.server_addr) - 1) {
```

Comment:
```suggestion
	if (k_sem_take(&probe_sem, HAWKBIT_SET_SERVER_TIMEOUT) == 0) {
		if (config->server_addr != NULL) {
			if (strnlen(config->server_addr, sizeof(hb_cfg.server_addr)) >= sizeof(hb_cfg.server_addr)) {
```

The ``>=`` could probably also be a ``==`` as the maximum val of the `strnlen()` would be `sizeof(hb_cfg.server_addr)`

Reply:
I think you'd still want to make sure that there is a null byte at the end of the buffer. So strnlen should return sizeof(hb_cfg.server_addr) - 1 at the most.